src/cr/cube/slices.py

```diff
@@ -0,0 +1,1140 @@
+        return tuple(values)
+
+    @lazyproperty
+    def _interleaved_base_values(self):
+        base_values = []
+        for i in range(len(self._base_vector.base_values)):
+            base_values.append(self._base_vector.base_values[i])
+            for inserted_vector in self._opposite_inserted_vectors:
+                if i == inserted_vector.anchor:
+                    insertion_value = np.sum(
+                        self._base_vector.base_values[inserted_vector.addend_idxs]
+                    )
+                    base_values.append(insertion_value)
+        return tuple(base_values)
+
+
+class _AssembledInsertionVector(_AssembledVector):
+    @lazyproperty
+    def anchor(self):
+        return self._base_vector.anchor
+
+
+class Calculator(object):
+    def __init__(self, assembler):
+        self._assembler = assembler
+
+    @lazyproperty
+    def row_proportions(self):
+        return np.array([row.proportions for row in self._assembler.rows])
+
+    @lazyproperty
+    def column_proportions(self):
+        return np.array([col.proportions for col in self._assembler.columns]).T
+
+    @lazyproperty
+    def table_proportions(self):
+        return np.array([row.table_proportions for row in self._assembler.rows])
+        # # return self.counts / self.table_base
+        # if not np.any(self.table_margin):
+        #     proportions = np.empty(self.counts.shape)
+        #     proportions[:] = np.nan
+        #     return proportions
+        # return self.counts / self.table_margin
+
+    @lazyproperty
+    def row_margin(self):
+        return np.array([row.margin for row in self._assembler.rows])
+
+    @lazyproperty
+    def column_margin(self):
+        return np.array([column.margin for column in self._assembler.columns])
+
+    @lazyproperty
+    def table_margin(self):
+        return self._assembler.table_margin
+
+    @lazyproperty
+    def row_base(self):
+        return np.array([row.base for row in self._assembler.rows])
+
+    @lazyproperty
+    def column_base(self):
+        return np.array([column.base for column in self._assembler.columns])
+
+    @lazyproperty
+    def table_base(self):
+        return self._assembler.table_base
+
+    @lazyproperty
+    def counts(self):
+        return np.array([row.values for row in self._assembler.rows])
+
+    @lazyproperty
+    def base_counts(self):
+        return np.array([row.base_values for row in self._assembler.rows])
+
+    @lazyproperty
+    def row_labels(self):
+        return tuple(row.label for row in self._assembler.rows)
+
+    @lazyproperty
+    def column_labels(self):
+        return tuple(column.label for column in self._assembler.columns)
+
+    @lazyproperty
+    def zscore(self):
+        # return self._calculate_std_res(
+        return self._scalar_type_std_res(
+            self.counts, self.table_margin, self.column_margin, self.row_margin
+        )
+
+    @staticmethod
+    def _scalar_type_std_res(counts, total, colsum, rowsum):
```

Comment:
This will go to CAT x CAT slice, and get computed at construction time of either columns or rows (but will still be a lazy property). By assigning it to vectors, we'll be able to naturally intersperse NaNs where needed, when assembling rows (or cols).